### PR TITLE
allow beacon_admin to accept opts to configure live_session.

### PR DESCRIPTION
**Pull request by xn**
Admin routes for a beacon site.

  ## Examples

      defmodule MyAppWeb.Router do
        use Phoenix.Router
        use Beacon.Router

        scope "/", MyAppWeb do
          pipe_through :browser
          beacon_admin "/admin", on_mount: [SomeHook]
        end
      end

  ## Options

    * `:on_mount` (optional) , an optional list of `on_mount` hooks passed to `live_session`.
    This will allow for authenticated routes, among other uses.

### CODE REVIEW

File: lib/beacon/router.ex [hunk 1]
```diff
@@ -153,6 +172,22 @@ defmodule Beacon.Router do
     end
   end
 
+  @doc false
+  def __admin_session_opts__(opts) do
+    if Keyword.has_key?(opts, :root_layout) do
```
Comment (leandrocp) [inline lib/beacon/router.ex:177]: Let's make `:layout` reserved as well.

**Review by leandrocp — CHANGES_REQUESTED** [review]
Nice one. Just one comment to make `:layout` reserved just like `:root_layout` and you'll need to import `beacon_admin/2` here https://github.com/BeaconCMS/beacon/blob/f4072b2e62fc0757df6186445e1b06f16df2951a/lib/beacon/router.ex#L21

File: lib/beacon/router.ex (hunk 1 above)
Comment (xn) [reply lib/beacon/router.ex:177]: I missed that. Thanks!

**Review by leandrocp — APPROVED** [review]
🎉

**Pull request merged**